src/morphodict/lexicon/util.py

```diff
@@ -0,0 +1,19 @@
+import unicodedata
+from unicodedata import normalize
+
+EXTRA_REPLACEMENTS = str.maketrans({"ł": "l", "Ł": "L", "ø": "o", "Ø": "O"})
+
+
+def to_source_language_keyword(s: str) -> str:
+    """Convert a source-language wordform to an indexable keyword
+
+    Currently removes accents, and leading and trailing hyphens.
+
+    There will be collisions but we could use edit distance to rank them.
+    """
+    return (
+        "".join(c for c in normalize("NFD", s) if unicodedata.combining(c) == 0)
+        .translate(EXTRA_REPLACEMENTS)
+        .strip("-")
+        .lower()
```

Comment:
`.lower()` and `.upper()` are not safe to apply on a per-character basis. See: https://unicode.org/Public/UNIDATA/SpecialCasing.txt

```suggestion
        "".join(c for c in normalize("NFD", s).lower() if unicodedata.combining(c) == 0)
        .translate(EXTRA_REPLACEMENTS)
        .strip("-")
```